File: source/tree/asset/header/display.go
package header

import "fmt"

func Display(instance *Header, leader string, final bool) {
	starter := ""
	if final {
		starter = "╰─ "
	} else {
		starter = "├─ "
	}

	fmt.Printf("%s%s%s\n", leader, starter, instance.Name)

	for headerName, _ := range instance.Headers {
		newLeader := ""
		if final {
			newLeader = leader + "   "
		} else {
			newLeader = leader + "│  "
		}

		fmt.Printf("%s\u001b[36m→\u001b[0m %s\n", newLeader, headerName)
	}

	for sourceName, _ := range instance.Sources {
		newLeader := ""
		if final {
			newLeader = leader + "   "
		} else {
			newLeader = leader + "│  "
		}

		fmt.Printf("%s\u001b[33m←\u001b[0m %s\n", newLeader, sourceName)
	}
}

File: test/example/filesystem/file/display.cpp
#include <ostream>
#include <string>

#include "file.hpp"

std::ostream &operator<<(std::ostream &stream, const File &file) {
    file.display(stream, "", false);
    return stream;
}

void File::display(std::ostream &stream,
        const std::string &leader,
        const bool &final) const {

    std::string starter = "";
    if(final)
        starter = "╰─ ";
    else
        starter = "├─ ";

    stream << leader << starter << this->name << '\n';
}
